resource/resource_graph.go

```diff
@@ -430,7 +410,7 @@ func (g *Graph) MergeAdd(toAdd *Graph) error {
 			return err
 		}
 		parents := toAdd.getAllChildrenOf(node)
```

Comment:
In the below `ReplaceNodesParents` function, I got freaked out by this `parents = getAllChildren` line. And I regret trying to fix that. So I left this the way it was.

Reply:
> And I regret trying to fix that.

Confused what you mean, as your changes look good below. I see this line in `ReplaceNodesParents`:  `otherChildren := other.getAllChildrenOf(node)`; did you not want to do a similar renaming here, so this variable is not called `parents`? I feel like `parents` and `parent` are incorrect names here.

Reply:
Maybe you're right and my changes are more correct than I give them credit for. I buy they are "locally" correct. But I was still getting confused on the bigger picture of what the algorithm is intending to do.

Reply:
> But I was still getting confused on the bigger picture of what the algorithm is intending to do.

I'm also still confused; I think there are more misusages of "parent" and "child" for both variable and function naming in the graph/manager (I'm highly suspicious of `markChildrenForUpdate` in particular). These changes do seem "locally" correct so definitely a step in the right direction.